src/contracts/base/ParaSwapLiquiditySwapAdapter.sol

```diff
@@ -0,0 +1,233 @@
+   * @param reserve The address of the asset
+   */
+  function renewAllowance(address reserve) public {
+    IERC20(reserve).safeApprove(address(POOL), 0);
+    IERC20(reserve).safeApprove(address(POOL), type(uint256).max);
+  }
+
+  /// @inheritdoc IParaSwapLiquiditySwapAdapter
+  function swapLiquidity(
+    LiquiditySwapParams memory liquiditySwapParams,
+    PermitInput memory collateralATokenPermit
+  ) external nonReentrant {
+    // Offset in August calldata if wanting to swap all balance, otherwise 0
+    if (liquiditySwapParams.offset != 0) {
+      (, , address aToken) = _getReserveData(liquiditySwapParams.collateralAsset);
+      uint256 balance = IERC20(aToken).balanceOf(liquiditySwapParams.user);
+      require(balance <= liquiditySwapParams.collateralAmountToSwap, 'INSUFFICIENT_AMOUNT_TO_SWAP');
+      liquiditySwapParams.collateralAmountToSwap = balance;
+    }
+
+    // true if flashloan is needed to swap liquidity
+    if (!liquiditySwapParams.withFlashLoan) {
+      _swapAndDeposit(liquiditySwapParams, collateralATokenPermit);
+    } else {
+      // flashloan of the current collateral asset
+      _flash(liquiditySwapParams, collateralATokenPermit);
+    }
+  }
+
+  /**
+   * @dev Executes the collateral swap after receiving the flash-borrowed assets
+   * @dev Workflow:
+   * 1. Sell flash-borrowed asset for new collateral asset
+   * 2. Supply new collateral asset
+   * 3. Pull aToken collateral from user and withdraw from Pool
+   * 4. Repay flashloan
+   * @param assets The addresses of the flash-borrowed assets
+   * @param amounts The amounts of the flash-borrowed assets
+   * @param premiums The premiums of the flash-borrowed assets
+   * @param initiator The address of the flashloan initiator
+   * @param params The byte-encoded params passed when initiating the flashloan
+   * @return True if the execution of the operation succeeds, false otherwise
+   */
+  function executeOperation(
+    address[] calldata assets,
+    uint256[] calldata amounts,
+    uint256[] calldata premiums,
+    address initiator,
+    bytes calldata params
+  ) external returns (bool) {
+    require(msg.sender == address(POOL), 'CALLER_MUST_BE_POOL');
+    require(initiator == address(this), 'INITIATOR_MUST_BE_THIS');
+
+    (
+      LiquiditySwapParams memory liquiditySwapParams,
+      PermitInput memory collateralATokenPermit
+    ) = abi.decode(params, (LiquiditySwapParams, PermitInput));
+
+    address flashLoanAsset = assets[0];
+    uint256 flashLoanAmount = amounts[0];
+    uint256 flashLoanPremium = premiums[0];
+
+    // sell the flashLoanAmount minus the premium, so flashloan repayment is guaranteed
+    // flashLoan premium stays in the contract
+    uint256 amountReceived = _sellOnParaSwap(
+      liquiditySwapParams.offset,
+      liquiditySwapParams.paraswapData,
+      IERC20Detailed(flashLoanAsset),
+      IERC20Detailed(liquiditySwapParams.newCollateralAsset),
+      flashLoanAmount - flashLoanPremium,
```

Comment:
imo this should use the `liquiditySwapParams.collateralAmountToSwap` instead of calculating the value.
Technically it should be the same, but imo it's a bit weird to use a "different" value then the one supplied by the user.

Reply:
after some discussion with @parth-15 i agree now that it's fine. Thx for clarification.